tools/cldr-code/src/main/java/org/unicode/cldr/tool/ChartPersonName.java

```diff
@@ -66,10 +65,10 @@ public String getFileName() {
 
     enum Filter {Main, Sorting, Monogram}
     enum Source {NativeSamples, ForeignSamples}
-
+ 
     @Override
-    public void writeContents(Writer pw) throws IOException {
-        CLDRFile cldrFile = FACTORY.make(locale, true);
+    public void writeContents(Writer pw, Factory factory) throws IOException {
```

Comment:
@macchiati  this is now the abstract function to override for adding a chart.

Reply:
ok. I think I need to go back through the other chart subclasses to make sure they are all kosher. Can do later (no point in generating charts when people can't see v43)

Reply:
I updated the subtask description with instructions.